.github/actions/BuildJlm/action.yml

```diff
@@ -0,0 +1,83 @@
+name: "Build jlm"
+description: "Configures and builds jlm along with the necessary dependencies."
+
+inputs:
+  enable-hls:
+    description: "Enables HLS dialect. Default is 'false'."
+    default: "false"
+    required: false
+
+  enable-mlir:
+    description: "Enables MLIR dialect. Default is 'false'."
+    default: "false"
+    required: false
+
+  enable-coverage:
+    description: "Enables computation of test coverage. Default is 'false'."
+    default: "false"
+    required: false
+
+  cxx:
+    description: "Set C++ compiler. Default is 'clang++'."
+    default: "clang++-16"
+    required: false
+
+  make-target:
+    description: "The make target that is invoked for compilation."
+    default: "all"
+    required: false
+
+runs:
+  using: "composite"
+  steps:
+    - name: "Validate input"
+      if: ${{(inputs.enable-hls != 'true' && inputs.enable-hls != 'false')
+        || (inputs.enable-mlir != 'true' && inputs.enable-mlir != 'false')
+        || (inputs.enable-coverage != 'true' && inputs.enable-coverage != 'false')}}
+      run: |
+        echo "Invalid composite action inputs:"
+        echo "enable-hls: ${{inputs.enable-hls}}"
+        echo "enable-mlir: ${{inputs.enable-mlir}}"
+        echo "enable-coverage: ${{inputs.enable-coverage}}"
+        exit 1
+      shell: bash
+
+    - name: "Install LLVM dependencies"
+      uses: ./.github/actions/InstallLlvmDependencies
+
+    - name: "Install HLS dialect dependencies"
+      if: ${{inputs.enable-hls == 'true'}}
+      uses: ./.github/actions/BuildCirct
+
+    - name: "Install MLIR dialect dependencies"
+      if: ${{inputs.enable-mlir == 'true'}}
+      uses: ./.github/actions/BuildMlirDialect
+
+    - name: "Collect command line arguments"
+      id: collect
+      run: |
+        export JLM_CONFIGURE_ARGUMENTS="--target release --enable-asserts"
+        if [[ "${{inputs.enable-hls}}" == "true" ]]; then
```

Comment:
Using bash if statements here and GitHub yaml if statements above is perhaps not ideal, but I can't come up with an alternative that avoids it, without creating a bunch of step outputs.

Reply:
I had the same problem. The ugliness is very contained and local. So I did not worry too much about it.